### PR TITLE
test: comprehensive coverage improvements (16 new tests + bug fix)

### PR DESCRIPTION
**Pull request by orieg**
## Summary

- **Bug fix**: `int_to_mixed_004.phpt` was a copy-paste of `int_to_int_004.phpt` — tested `Judy::INT_TO_INT` instead of `Judy::INT_TO_MIXED`. Fixed type, added MIXED SKIPIF guard, and changed values to strings.
- **16 new tests** filling coverage gaps identified during issue #43 planning:
  - `memoryUsage()` — zero existing tests; now 2 (basic behavior for INT_TO_INT/BITSET + all 5 types including STRING_TO_* returning null)
  - `free()` lifecycle — 2 tests (reset state verification + re-populate after free)
  - Constructor error — invalid type (999, -1, 0) throws exception
  - Clone independence — 4 tests for INT_TO_INT, INT_TO_MIXED, STRING_TO_INT, STRING_TO_MIXED (only BITSET had coverage before)
  - Boundary values — 4 tests (PHP_INT_MAX key/value, 1000-char string keys, empty string key, special chars/unicode/emoji)
  - Batch edge cases — 2 tests (empty input arrays, 1000+ element operations)
  - Increment boundary — large amounts, zero amount, accumulation, decrement

**Test count**: 99 → 115 (all passing)

Ref: #43

## Test plan
- [x] All 115 tests pass in Docker (PHP 8.3)
- [ ] CI matrix (PHP 8.1-8.5, Linux + Windows) passes

### CODE REVIEW

File: tests/boundary_long_string_001.phpt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+$j[$key_a] = 1;
+$j[$key_b] = 2;
+echo "500-char key 'a': " . $j[$key_a] . "\n";
+echo "500-char key 'b': " . $j[$key_b] . "\n";
+
+// Single char key alongside long keys
+$j["x"] = 99;
+echo "Single char key: " . $j["x"] . "\n";
+
+echo "Count: " . $j->count() . "\n";
```
[the remark this answers is not in the document]
Comment (orieg) [reply tests/boundary_long_string_001.phpt]: Agreed — added tests for max allowed key length (65535 bytes) and oversized key (65536 bytes, throws exception) in 48a5196. This validates the `PHP_JUDY_MAX_LENGTH` guard in `write_dimension_helper`.
